### PR TITLE
Improve check for dangerous links...

### PR DESCRIPTION
**Pull request by sgiehl**
and some other UI test related stuff

### CODE REVIEW

File: tests/lib/screenshot-testing/support/chai-extras.js [hunk 1]
```diff
@@ -501,14 +501,37 @@ function performAutomaticPageChecks(done) {
...
+            var linkElements = document.getElementsByTagName('a');
+
+            elements:
+            for (var i = 0; i !== linkElements.length; ++i) {
+                var element = linkElements.item(i);
+
+                var href = element.getAttribute('href');
+                if (/^(javascript|vbscript|data):;*[^;]+/.test(href)) {
+                    for (var index in safeJsLinks) {
+                        if (safeJsLinks[index].test(href)) {
```
Comment (fdellwing) [inline tests/lib/screenshot-testing/support/chai-extras.js]: Is there any reason why you keep using regex here? Wouldn't it be easier to just test via `Array.prototype.indexOf()`? You are using `^` and `$` so, if you do not plan to use wildcards in safe links any time soon, it would be performanter to use index of.

**Comment by fdellwing** [comment]
It might be better to run `Array.prototype.indexOf()` instead of `RegExp.prototype.test()`: https://jsperf.com/matomo-org-matomo-13731/1

But I have to admit, the difference is very minor on my system. And the performance gain will cannibalize itself, as soon as you use more wildcards.

File: plugins/CorePluginsAdmin/tests/UI/TagManagerTeaser_spec.js [hunk 2]
```diff
@@ -82,11 +82,11 @@ describe("TagManagerTeaser", function () {
     });
 
     it('should be possible to disable page and redirect to home', function (done) {
-        capturePage(done, 'admin_page_disable', function (page) {
+        expect.page("").contains("#dashboard", function (page) {
```
Comment (tsteur) [inline plugins/CorePluginsAdmin/tests/UI/TagManagerTeaser_spec.js]: Is that change on purpose?

Comment (tsteur) [reply plugins/CorePluginsAdmin/tests/UI/TagManagerTeaser_spec.js]: there is no screenshot name etc... and shouldn't it capture?

Comment (sgiehl) [reply plugins/CorePluginsAdmin/tests/UI/TagManagerTeaser_spec.js]: yes. it now only checks if the loaded page has the element `#dashboard`. Imho there is no need to compare the actual content as a screenshot

**Comment by sgiehl** [comment]
closing in favor of #13741
